#3178771 drupal 9 compatibility

krystalcode opened the pull request:
Drupal.org issue: https://www.drupal.org/project/commerce_price_rule/issues/3178771

To use this branch add the following to the `repositories` section in your `composer.json` file:
```
"repositories": {
     "commerce-price-rule": {                                                                                                                                                                                                         
           "type": "vcs",                                                                                                                                                                                                               
            "url": "https://github.com/krystalcode/drupal8-commerce-price-rule"                                                                                                                                                          
     },
}
```
And then:
```
"require": {
    "drupal/commerce_price_rule": "dev-3178771-drupal-9-compatibility
}
```

krystalcode committed:
Minimum version of commerce 2.17 that supports Drupal 9

krystalcode committed:
Updated info file keys for Drupal 9 compatibility

krystalcode committed:
Replaced deprecated entity manager with entity type manager

barinder committed:
Fixed dependenciy, added commerce_order

barinder committed:
Drupal 9 API changes

krystalcode commented on commerce_price_rule.info.yml:
```diff
@@ -2,7 +2,7 @@ name: Commerce Price Rule
 dependencies:
   - commerce:commerce_product
-  - system (>=8.3.0)
+  - commerce:commerce_order
```
**CHANGE REQUEST**

@barinder Were you getting a particular error? As far as I see the module does not depend on orders.

barinder replied:
@krystalcode Yes, if commerce_order is not enabled following fatal error is thrown while adding price rule.
`"Drupal\Component\Plugin\Exception\PluginNotFoundException: The "commerce_order_item" entity type does not exist. in Drupal\Core\Entity\EntityTypeManager->getDefinition() (line 150 of /Users/barinder/Sites/d9/core/lib/Drupal/Core/Entity/EntityTypeManager.php)."`

krystalcode replied:
@barinder Ok I just got some time to look into this. This is a bug in the Commerce module; I have created an [issue](https://www.drupal.org/project/commerce/issues/3184470) and it should be fixed there. We should not be demanding the order module here.

krystalcode committed:
Removed again dependency on the Commerce Order module

krystalcode committed:
Corrected variable name

krystalcode committed:
Updated calculation plugin interface

ConfigurablePluginInterface has been replaced in Core by ConfigurableInterface
and DependentPluginInterface. It was previously replaced only with
ConfigurableInterface, added DependentPluginInterface as well.

krystalcode commented:
Merged in Drupal.org.
